docs/README.md

```diff
@@ -194,43 +205,92 @@ TEST(add_basic) {
 TEST_MAIN()
 ```
 
-# Write Unit Tests for `slideright()` and `flip()` (Lab 2)
+# Writing Unit Tests for `slideright()` and `flip()` array functions
+
+Now, let's add some real test cases for `slideRight()` and `flip()` to `arrays_tests.cpp`.
+
+For example, here's one test for each function (you can replace your existing code in `arrays_tests.cpp` with the code below):
+
+```c++
+#include "arrays.h"
+#include "unit_test_framework.h"
+
+// A helper function for comparing arrays. Returns true if the
+// arrays are the same size and contain the same values.
+bool compare_arrays(int arr1[], int size1, int arr2[], int size2) {
+  if (size1 != size2) {
+    return false;
+  }
+
+  for (int i = 0; i < size1; ++i) {
+    if (arr1[i] != arr2[i]) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+TEST(test_slide_right_1) {
+  int testing[] = { 4, 0, 1, 3, 3 };
+  int correct[] = { 3, 4, 0, 1, 3 };
+  slideRight(testing, 5);
+  ASSERT_TRUE(compare_arrays(testing, 5, correct, 5));
+}
+
+TEST(test_flip_1) {
+  int testing[] = { 4, 0, 1, 3, 3 };
+  int correct[] = { 3, 3, 1, 0, 4 };
+  flip(testing, 5);
+  ASSERT_TRUE(compare_arrays(testing, 5, correct, 5));
+}
 
-Add test cases for `slideRight()` and `flip()` to `lab02_tests.cpp`.
+TEST_MAIN() // No semicolon!
+```
 
 Compile and run the tests with the following commands:
 
 ```console
-$ g++ -Wall -Werror -pedantic -O1 -std=c++11 lab02.cpp lab02_tests.cpp -o lab02_tests.exe
-$ ./lab02_tests.exe
+$ g++ -Wall -Werror -pedantic -O1 -std=c++11 arrays.cpp arrays_tests.cpp -o arrays_tests.exe
+$ ./arrays_tests.exe
 ```
 
-Of course, if you run your tests against your own (correct) implementations
-in lab02.cpp, the tests should all pass. (We call tests that pass when run
-against a correct implementation _valid_.) You can temporarily change one of
-your functions to contain a bug and observe that some tests fail.
+A failed test indicates there's a bug in our `arrays.cpp` code. The tests above caught
+the bug in `slideRight` but not the one in `flip` (recall we mentioned there is a bug in both).
+You'll need to write more tests to create a thorough testing suite!
+Ideally, you should have enough tests that any reasonable bug will cause at least
+one test to fail.
 
-Once you feel your tests are thorough, submit `lab02_tests.cpp` to the lab 2
+On the other hand, when run against a correct implementation, all of your tests should pass. (Otherwise, it would be giving a false positive for detecting a bug.) If a test passes on a correct implementation,
+we call it _valid_.
+
+Once you feel your tests are thorough, submit `arrays_tests.cpp` to the unit test framework tutorial
 autograder. It will check your tests against a set of buggy implementations
```

Comment:
```suggestion
Once you feel your tests are thorough, submit `arrays_tests.cpp` to the unit test framework tutorial
[autograder](https://autograder.io/). It will check your tests against a set of buggy implementations
```